apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts

```diff
@@ -77,4 +117,31 @@ export class EventTypesAtomService {
 
     return eventType.eventType;
   }
+
+  async checkCanUpdateTeamEventType(userId: number, eventTypeId: number, teamId: number, scheduleId: number) {
+    await this.checkTeamOwnsEventType(userId, eventTypeId, teamId);
+    await this.teamEventTypeService.validateEventTypeExists(teamId, eventTypeId);
+    await this.eventTypeService.checkUserOwnsSchedule(userId, scheduleId);
+  }
+
+  async checkTeamOwnsEventType(userId: number, eventTypeId: number, teamId: number) {
+    const membership = await this.dbRead.prisma.membership.findFirst({
+      where: {
+        userId,
+        teamId,
+        accepted: true,
+        OR: [{ role: "ADMIN" }, { role: "OWNER" }],
+      },
+      select: {
+        team: {
+          select: {
+            eventTypes: true,
+          },
+        },
+      },
+    });
+    if (!membership?.team?.eventTypes?.some((item) => item.id === eventTypeId)) {
+      throw new ForbiddenException(`Team with ID=${teamId} does not own event type with ID=${eventTypeId}`);
```

Comment:
update error message to reflect that users with role MEMBER are not allowed to access the endpoint

Reply:
https://github.com/calcom/cal.com/pull/17145/commits/22aaa94efb88ccdf3d41624ca371ec759f04cbe2